src/main/java/com/terraformersmc/campanion/block/LeatherTanner.java

```diff
@@ -99,10 +101,33 @@ public boolean hasRandomTicks(BlockState state) {
 	}
 
 	@Override
-	public void scheduledTick(BlockState state, ServerWorld world, BlockPos pos, Random random) {
-		super.scheduledTick(state, world, pos, random);
+	public void randomTick(BlockState state, ServerWorld world, BlockPos pos, Random random) {
 		if (getAge(state) == 1) {
-			if (random.nextInt(8) == 0) {
+
+			int speedMultiplier = 1;
+			Direction facing = state.get(FACING);
+			ArrayList<BlockState> blockStates = new ArrayList<>();
+
+			//Check if there is a lit campfire within the immediate surrounding blocks and on the flat face of the tanner
+			blockStates.add(world.getBlockState(pos.east().north()));
+			blockStates.add(world.getBlockState(pos.east().south()));
+			blockStates.add(world.getBlockState(pos.west().north()));
+			blockStates.add(world.getBlockState(pos.west().south()));
+			if (facing == NORTH || facing == SOUTH) {
+				blockStates.add(world.getBlockState(pos.south()));
+				blockStates.add(world.getBlockState(pos.north()));
+			} else if (facing == EAST || facing == WEST){
+				blockStates.add(world.getBlockState(pos.west()));
+				blockStates.add(world.getBlockState(pos.east()));
+			}
+
+			for (BlockState currState : blockStates) {
+				if (currState.getBlock() instanceof CampfireBlock && currState.get(LIT)) {
+					speedMultiplier *= 1.5;
+				}
+			}
+
+			if (random.nextInt(8 / speedMultiplier) == 0) {
```

Comment:
Maybe we should consider using `nextFloat() <= speedMultiplier / 8`? That would give the following chances:

* no campfire: 12.5%
* 1 campfire: 18.75%
* 2 campfires: 28.13%
* 3 campfires: 42.19%
* 4 campfires: 63.28%
* 5 campfires: 94.92%
* 6 campfires: 100%